src/human_player_module.c

```diff
@@ -0,0 +1,391 @@
+{
+    int size = sizeof CARD_VALUES / sizeof *CARD_VALUES;
+
+    for (int i = 0; i < size; ++i)
+    {
+        if (!strcmp(CARD_VALUES[i], entered_value))
+        {
+            return false;
+        }
+    }
+    return true;
+}
+
+/**
+ * @brief Records the Human player input during his turn and performs appropriate functions 
+ * Called each time when it's the human player's turn 
+ * Keyboard Input from Human 
+ *        Q/q - Exit game
+ *        R/r - Requests card 
+ *        E/e - End Turn
+ *        Other- Map to appropriate Card Info
+ */
+
+error_e record_human_input()
+{
+    char user_input[2];
+    printf("Please enter your choice \n");
+    scanf("%s", user_input);
+
+    if (user_input[0] == 'q' || user_input[0] == 'Q')
+    {
+        quit_game();
+    }
+    else if (user_input[0] == 'r' || user_input[0] == 'R')
+    {
+        request_card(HUMAN, 1);
+    }
+
+    else if (user_input[0] == 'e' || user_input[0] == 'E')
+    {
+        if (g_card_requested)
+        {
+            end_turn(g_player_on_turn);
+        }
+        else
+        {
+            printf("Please draw a card before you can end your turn");
+            display_player_deck(human_player);
+        }
+    }
+    else
+    {
+        if (validate_card(user_input))
+        {
+            //To be added if the format of user input is different
+            Card_t *human_card_choice = map_user_input(user_input);
+            //If valid add the card to the discard_pile
+            if (check_is_valid_turn(&g_card_on_table, human_card_choice))
+            {
+                Deck_t **pp_head = &(human_player->cards_on_hand);
+
+                Card_t *to_remove_card = pick_card_from_deck(pp_head, human_card_choice);
+
+                if (to_remove_card != NULL)
+                {
+                    ERR_RET(remove_card_from_deck(pp_head, human_card_choice));
+                    add_card_at_end(g_discard_pile, *to_remove_card);
+                }
+            }
+        }
+        else
+        {
+            invalid_turn_warning();
+        }
+    }
+    return ERR_SUCCESS;
+}
+/**
+ * @brief Remove a particular card from the given deck
+ * Can be used to discard a particular card - during each turn
+ * Works for both the computer as well as human deck of cards
+ * 
+ * @param pp_head 
+ * @param card_to_be_removed 
+ * @return error_e 
+ */
+error_e remove_card_from_deck(Deck_t **pp_head, Card_t *card_to_be_removed)
+{
+    Deck_t *to_delete = *pp_head;
+    Deck_t *prev = *pp_head;
+    if (*pp_head == NULL)
+    {
+        printf("List is already empty.");
+        return ERR_EMPTY_LIST;
+    }
+    while (to_delete->next != NULL)
+    {
+        if (to_delete->card.color == card_to_be_removed->color && to_delete->card.name == card_to_be_removed->name)
+        {
+            prev->next = to_delete->next;
+            free(&(to_delete->card));
+            return true;
+        }
+        to_delete = to_delete->next;
+    }
+    return false;
```

Comment:
Same as the above comment: The function return is error_e enum type. Why it returns boolean here?